### PR TITLE
docs: update doc

### PR DESCRIPTION
**Pull request by shiqi**
1.  correct the ascProvider parameter doc: this is not necessarily the team id, it should be the team short name as stated in the apple developer forum. https://forums.developer.apple.com/thread/113798 and I've tested this and proved that for our app the team id doesn't work because our team short name is different than our team id. 

2. adding some preliminary doc about signing requirement of harden-runtime and the entitlement, looking for suggestions on that part.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,7 +42,10 @@ For notarization, you need the following things:
 1. Xcode 10 or later installed on your Mac.
 2. An [Apple Developer](https://developer.apple.com/) account.
 3. [An app-specific password for your ADC account’s Apple ID](https://support.apple.com/HT204397).
-
+4. Your app may need to be signed with harded-runtime and the following entitlements:
+    1. com.apple.security.cs.allow-jit
+    1. com.apple.security.cs.allow-unsigned-executable-memory
+    1. com.apple.security.cs.allow-dyld-environment-variables
```
Comment (Kilian) [inline README.md]: I think you only really need `com.apple.security.cs.allow-unsigned-executable-memory` for a vanilla Electron to function so would prefer to keep this list as minimal as possible, but possibly provide a link to a [list of other entitlements](https://developer.apple.com/documentation/bundleresources/entitlements) that specific apps might need.

Comment (MarshallOfSound) [reply README.md]: @Kilian `allow-jit` is definitely required if you want V8 to run it's optimizations 🤔

Comment (MarshallOfSound) [reply README.md]: Not sure about the dyld environment variables, seems app specific there

Comment (shiqi) [reply README.md]: seems this is subject to the app, how about we just mention it with the link of all the entitlement here?

Comment (shiqi) [reply README.md]: @Kilian kindly bump this PR, let me know your thoughts on this. 🙇

File: README.md [hunk 2]
```diff
@@ -42,10 +42,7 @@ For notarization, you need the following things:
 1. Xcode 10 or later installed on your Mac.
 2. An [Apple Developer](https://developer.apple.com/) account.
 3. [An app-specific password for your ADC account’s Apple ID](https://support.apple.com/HT204397).
-4. Your app may need to be signed with harded-runtime and the following entitlements:
-    1. com.apple.security.cs.allow-jit
-    1. com.apple.security.cs.allow-unsigned-executable-memory
-    1. com.apple.security.cs.allow-dyld-environment-variables
+4. Your app may need to be signed with hardened-runtime and [list of entitlements](https://developer.apple.com/documentation/bundleresources/entitlements)
```
Comment (Kilian) [inline README.md]: You definitely need at least `com.apple.security.cs.allow-unsigned-executable-memory` and possibly `com.apple.security.cs.allow-jit` (Though I remain unconvinced: https://github.com/electron/electron-osx-sign/issues/188#issuecomment-483778579) so not mentioning those would be unkind.

The `allow-jit` one at least does no harm (especially if it's superseded by/included in `allow-unsigned-executable-memory`), so lets keep it at those two.

File: README.md (hunk 1 above)
Comment (Kilian) [reply README.md]: (closing this because outdated)

**Pull request merged**